app/routes/screens/main/clubFeed/ClubFeed.js

```diff
@@ -44,23 +47,9 @@ class ClubFeed extends Component {
 
   render() {
     const { club, clubMembers, activities } = this.props;
-    if (club === Loading)
-      return (
-        <View style={styles.container}>
-          <Icon name="cached" color="#FC4C02" size={50} />
-          <Text style={styles.text}>fetching data</Text>
-        </View>
-      );
+    if (club === IsLoading) return <Loading />;
 
-    if (isFaulty(club))
-      return (
-        <View style={styles.container}>
-          <Icon name="error" color="#FC4C02" size={100} />
-          <Text style={styles.text}>
-            Oops, I did it again: {getDefect(club)}
-          </Text>
-        </View>
-      );
+    if (isFaulty(club)) return <Faulty message={getDefect(club)} />;
```

Comment:
i am considering using recompose https://github.com/acdlite/recompose
to manage component composition
i have to consider slots in this too
i started to do something about theming (light theme/dark theme) and feature flipping
the global thème is how do we compose correctly.